include/deal.II/grid/manifold_lib.h

```diff
@@ -503,6 +503,60 @@ class TorusManifold : public ChartManifold<dim,3,3>
   double r, R;
 };
 
+
+
+/**
+ * A mapping class that smoothly transitions from different manifold
+ * descriptions on different faces by a so-called transfinite
+ * interpolation. As opposed to setting the manifold description only on the
+ * boundary of a mesh, this approach makes sure that the deformation of the
+ * mesh according to a curved boundary description is applied to more than one
+ * element layer once the mesh gets refined, which is necessary to get
+ * high-order convergence rates (without this class, the error on cells close
+ * to the boundary does not exceed 3 on a circle).
+ *
+ * @note Only instantiated for dim > 1 as a transfinite mapping cannot be
+ * defined in 1D.
+ *
+ * @author Luca Heltai, Martin Kronbichler, 2017
+ */
+template <int dim, int spacedim=dim>
+class TransfiniteInterpolationManifold : public ChartManifold<dim,spacedim,dim>
+{
+public:
+  void initialize (const typename Triangulation<dim,spacedim>::cell_iterator &cell);
```

Comment:
I think you should also mention that the function evaluates the vertices and manifolds *at the time this function is called with a cell*, rather than saving the cell iterator and looking things up later on. This, for example, doesn't allow for meshes that change, and requires setting manifold ids on the faces *before* calling this function.